pkg/core/interop/contract/call.go

```diff
@@ -69,6 +69,20 @@ func Call(ic *interop.Context) error {
 		return fmt.Errorf("method not found: %s/%d", method, len(args))
 	}
 	hasReturn := md.ReturnType != smartcontract.VoidType
+
+	if ic.SaveInvocations {
+		var (
+			arrCount  = len(args)
+			truncated = false
+			argBytes  []byte
+		)
+		if argBytes, err = ic.DAO.GetItemCtx().Serialize(stackitem.NewArray(args), false); err != nil {
```

Comment:
I've found that `argBytes` gets modified by the time `EncodeBinaryWithContext` of `ContractInvocation` is called (when persisting to storage). Specifically, the type changes from an `Array` to a `Struct`. 

If I replace `ic.DAO.GetItemCtx()` with `stackitem.NewSerializationContext()` the content of `argBytes` is preserved. I'm not sure what the implications of this are but wanted to point this out as I find this somewhat worrisome. @AnnaShaleva @roman-khimov 

I found this out by a type assertion failure in `MarshallJSON()` of `ContractInvocation`, specifically at this line
```go
item, err = stackitem.ToJSONWithTypes(si.(*stackitem.Array))
```

Reply:
It's an unexpected behaviour, let me check.

Reply:
I found this specifically happens when trying to read the following T5 log (it's in block 2872 iirc so you don't have to sync much for it)
```
curl --location 'http://127.0.0.1:20332' \
--header 'Content-Type: text/plain' \
--data '{
  "jsonrpc": "2.0",
  "method": "getapplicationlog",
  "params": ["0x2f7b0f78df942212334c0b6c03239a1afc4518aeb4461cb5b77b31aaa9e0c1a5"],
  "id": 1
}
'
```

Reply:
Ref. https://github.com/nspcc-dev/neo-go/pull/3569#discussion_r1926620094.